pgmgr/dump_config.go

```diff
@@ -0,0 +1,110 @@
+package pgmgr
+
+import "strings"
+
+// DumpConfig stores the options used by pgmgr's dump tool
+// and defers connection-type options to the main config file
+type DumpConfig struct {
+	// exclusions
+	ExcludeSchemas []string `json:"exclude-schemas"`
+	ExcludeTables  []string `json:"exclude-tables"`
+	ExcludeData    []string `json:"exclude-data-tables"`
+
+	// inclusions
+	IncludeSchemas []string `json:"include-schemas"`
+	IncludeTables  []string `json:"include-tables"`
+
+	// options
+	DumpCompression string `json:"compress"`
+	DumpFile        string `json:"dump-file"`
+}
+
+// GetDumpFileRaw returns the literal dump file name as configured
+func (config DumpConfig) GetDumpFileRaw() string {
+	return config.DumpFile
+}
+
+// GetDumpFile returns the true dump file name
+// with or without the specified compression suffix
+func (config DumpConfig) GetDumpFile() string {
+	if config.IsCompressed() {
+		return config.DumpFile + ".gz"
+	}
+	return config.DumpFile
+}
+
+// IsCompressed returns the configured value of the DumpCompression flag.
+// Since compression is really beneficial to apply, we greedily
+// set to true for any string value other than "f"
+func (config DumpConfig) IsCompressed() bool {
+	return config.DumpCompression != "f"
+}
+
+func (config *DumpConfig) applyArguments(ctx argumentContext) {
+	if sliceValuesGiven(ctx, "exclude-schemas") {
+		config.ExcludeSchemas = ctx.StringSlice("exclude-schemas")
+	}
+	if sliceValuesGiven(ctx, "exclude-tables") {
+		config.ExcludeTables = ctx.StringSlice("include-tables")
+	}
+	if sliceValuesGiven(ctx, "exclude-table-data") {
+		config.ExcludeData = ctx.StringSlice("exclude-table-data")
+	}
+	if sliceValuesGiven(ctx, "include-schemas") {
+		config.IncludeSchemas = ctx.StringSlice("include-schemas")
+	}
+	if sliceValuesGiven(ctx, "include-tables") {
+		config.IncludeTables = ctx.StringSlice("include-tables")
+	}
+	if sliceValuesGiven(ctx, "seed-tables") {
+		deprecatedDumpFieldWarning("seed-tables", "include-tables", "command line arg")
```

Comment:
Actually, `seed-tables` and `include-tables` are not equivalent. `seed-tables` would line up with an `include-table-data` flag, which I don't see as part of this PR. That is, include/exclude tables refers to tables which are having their structures dumped; the old pgmgr behavior was to dump everything and then just the data for `seed-tables`.

In other words, if I wanted to configure pgmgr to dump my database but exclude tables X Y and Z entirely and only seed tables A B and C, is that possible?

Reply:
Good catch, this change would make it possible to exclude X Y and Z but not possible to fetch _only_ the data for A B and C. Unfortunately there is no flag for grab _only_ the data from specific tables -- we would still have to use `--data-only` which affects the whole dump.

I'll reintroduce (and alias) `seed-tables` as `data-only-tables` and rename `exclude-table-data` to `schema-only-tables`. to support this change I would need to also add the functionality back in that dumps twice: once with `--data-only` and once without it

Reply:
We discussed this further -- we'll try to boil all of these options down to just `exclude-schemas` and `seed-tables`. A `--schema-only` dump will be performed with all `exclude-schemas` omitted, and a `--data-only` dump will be performed with only `seed-tables` tables included (as the current version already does)